src/tdastro/obstable/roman_obstable.py

```diff
@@ -0,0 +1,249 @@
+        The angular radius of the observations (in degrees).
+    """
+
+    # Default column names for the ZTF survey data.
+    _default_colnames = {"dec": "DEC", "filter": "BANDPASS", "ra": "RA", "zp": "zp_nJy", "time": "time"}
+
+    # Default survey values.
+    _default_survey_values = {
+        "dark_current": _roman_dark_current,
+        "pixel_scale": ROMAN_PIXEL_SCALE,
+        "survey_name": "Roman",
+        "radius": np.sqrt(_roman_fov / np.pi),
+        "zodi_level": _roman_zodi_level_factor,
+    }
+
+    def __init__(self, table, colmap=None, ma_table_path=None, **kwargs):
+        colmap = self._default_colnames if colmap is None else colmap
+
+        self.apt_table = table
+        self.ma_table = _get_ma_table(ma_table_path)
+        roman_char = _get_roman_char()
+        self.zp_table = roman_char["zp_table"]
+        self.psf_table = roman_char["psf_table"]
+        self.thermal_table = roman_char["thermal_table"]
+        self.zodiacal_table = roman_char["zodiacal_min_table"]
+        self._append_apt_table()
+
+        super().__init__(self.apt_table, colmap=colmap, **kwargs)
+
+    def _append_apt_table(self):
+        self.apt_table["zp_abmag"] = 0.0
+        self.apt_table["N_Eff_Pix"] = 0.0
+        self.apt_table["zodi_countrate_min"] = 0.0
+        self.apt_table["thermal_countrate"] = 0.0
+        self.apt_table["time"] = 0.0
+        self.apt_table["exptime"] = 0.0
+
+        for f in np.unique(self.apt_table.BANDPASS):
+            zp_abmag = self.zp_table.loc[self.zp_table.element == f, "ABMag"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "zp_abmag"] = zp_abmag
+            n_eff_pix = self.psf_table.loc[self.psf_table["Filter"] == f, "N_Eff_Pix"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "N_Eff_Pix"] = n_eff_pix
+            sigma_zodi_min = self.zodiacal_table.loc[self.zodiacal_table["filter"] == f, "rate"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "zodi_countrate_min"] = sigma_zodi_min
+            sigma_thermal = self.thermal_table.loc[self.thermal_table["filter"] == f, "rate"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "thermal_countrate"] = sigma_thermal
+
+        for mat_number in np.unique(self.apt_table.MA_TABLE_NUMBER):
+            exptime = self.ma_table.loc[self.ma_table["MATableNumber"] == mat_number, "Exptime"].values[0]
+            self.apt_table.loc[mat_number == self.apt_table.MA_TABLE_NUMBER, "exptime"] = exptime
+
+        npass = len(np.unique(self.apt_table.PASS))
+        self.start_time = 61304.0
+        self.cadence = 5.0
+        times = self.start_time + self.cadence * np.arange(npass)
+        for i, pass_num in enumerate(np.unique(self.apt_table.PASS)):
+            self.apt_table.loc[pass_num == self.apt_table.PASS, "time"] = times[i]
+
+        self.apt_table["zp_nJy"] = mag2flux(self.apt_table["zp_abmag"])
+
+    def readnoise_func(self, exptime):
```

Comment:
If you wanted to, you could at an `@cite_function` decorator to pick up the reference in the docstring automatically. (You'd also need `from citation_compass import cite_function` at the top).